pyugrid/util.py

```diff
@@ -49,3 +49,28 @@ def _signed_area_tri(points):
     x3, y3 = points[2]
 
     return(((x1-x3)*(y2-y3)-(x2-x3)*(y1-y3))/2)
+
+def asarraylike(obj):
+    """
+    tests if obj acts enough like an array to be used in pyugrid. 
+
+    If it does the object is returned as is. If not, then numpy's
+    array() will be called on it.
+
+    This should catch netCDF4 variables, etc.
+
+    Note: these won't check if the attributes required actually work right.
+
+    :param obj: The object to check if it's like an array
+
+    """
+
+    must_have = ['dtype', 'shape']
```

Comment:
You mean one of [dask, biggus, ???] -- I don't think so.

At this point all we need is netCDF4 variable objects (which have not been tested -- we are using them, and the tests all pass, but I haven't confirmed that they are supporting lazy loading the way we want.

But the goal was to really do duck-typing -- anything that works is OK to use. So I only wanted to check for that attributes that we actually use. Sure, any number of things could pass through and fail later, but it's either that or specifically testing for types we know about -- which I think we should not do.

So I say one of:
 - what I"ve got now, with new methods added as we see the need
or
 - just go with __array__ and be done with it.

Or both -- now that I think about, __array__ means it can be turned into an ndarray (maybe efficiently), but it very well might not have any of the usual indexing, etc. operations.

So we need the attribute checks.